### DISCUSSION TITLE
Citation picker in source mode with VS Code and Zotero

### DISCUSSION THREAD

**Discussion by rpchastain** [quarto-dev/quarto-cli#9173] [Q&A] [answered]
### Description

I'm writing a Quarto manuscript in VS Code and using Zotero to manage my references. 

I love that in visual mode I have access to all my references just by typing `@`. However, in visual mode I lose syntax highlighting, and syntax highlighting is much more useful for me in general than the citation auto-picker. For that reason I usually work in source mode. 

It's easy enough to switch between source mode and visual mode when I need to insert a new reference. Unfortunately when switch from source mode to visual mode Quarto does a lot of automatic formatting for me, which ends up completely wrecking the format of my paper.

I'd love a way to have citation auto-picker available in source mode, but that doesn't seem like it will ever happen. As an alternative, I'd be satisfied if there was some way to stop Quarto from reformatting my document when I switch between source and visual editors. In lieu of that, is there a way to add decent syntax highlighting to visual mode?

Thanks!

**Comment by mcanouil** [comment, marked as answer]
This repository is about Quarto CLI.

Please look for an already open issues in the Quarto VSCode extension repository: https://github.com/quarto-dev/quarto


> As an alternative, I'd be satisfied if there was some way to stop Quarto from reformatting my document when I switch between source and visual editors.

That's not possible as several markdown syntaxes can lead to the same visual display.
Conversion is done by Pandoc, thus follows Markdown standard as defined/used by Pandoc.
Internally, the only thing that the Visual Editor is aware of is the Pandoc representation of the content, namely AST. The AST is used to go from one mode to the other.
